packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -29,9 +41,23 @@ export type QueuedRequestControllerEvents =
 export type QueuedRequestControllerActions =
   QueuedRequestControllerEnqueueRequestAction;
 
+type AllowedActions =
+  | NetworkControllerGetStateAction
+  | NetworkControllerSetActiveNetworkAction
+  | NetworkControllerSetProviderTypeAction
+  | NetworkControllerGetNetworkClientByIdAction
+  | NetworkControllerFindNetworkClientIdByChainIdAction
+  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
+  | AddApprovalRequest;
+
+export type QueuedRequestMessenger = ControllerMessenger<
+  QueuedRequestControllerActions | AllowedActions,
+  QueuedRequestControllerEvents
+>;
+
 export type QueuedRequestControllerMessenger = RestrictedControllerMessenger<
   typeof controllerName,
-  QueuedRequestControllerActions,
+  QueuedRequestControllerActions | AllowedActions,
   QueuedRequestControllerEvents,
   string,
```

Comment:
Shouldn't this be `AllowedActions['type']`?

Reply:
The type params for `RestrictedControllerMessenger` are as follows: 

1. Namespace for internal actions/events (i.e. controller name).
2. All usable actions, both internal and external to the controller.
3. All usable events, both internal and external to the controller.
4. Type names of external usable actions.
5. Type names of external usable events. 

This line is passing type param no. 2, so it should extend `ActionConstraint`, not `string`.

Reply:
Ah, sorry, I wasn't clear.

I was wondering whether we should change 4 and 5 here from `string` to something else. I guess we don't have to do that here, but if we're updating the type parameters is it worth it to make all of them accurate?

Reply:
Oh my mistake.😅 `string` and `never` are used in those slots for allow-all and deny-all, and usually `AllowedActions['type']` would be the right type argument, but in this case we get errors unless we pass in `QueuedRequestControllerActions['type']` or a wider type. 

The `QueuedRequestControllerMessenger` and `QueuedRequestController` are being used a bit differently than other controllers. It seems to be related to the `ExampleControllerActionTypes` and `ExampleControllerEventTypes` pattern that was introduced in `SelectedNetworkController`. I'll see if I can use those so we're using `RestrictedControllerMessenger` everywhere in this package, which I think will resolve the discrepancies.

Reply:
`AllowedActions['type']` now works here thanks to the updates introduced in https://github.com/MetaMask/core/pull/2050.

Fixed here: https://github.com/MetaMask/core/pull/1970/commits/1c2c4fce7ea5ccfea7d484b120f2012abc8afb57.